### PR TITLE
DAOS-2603 doc: Update security model documentation

### PR DESCRIPTION
**Pull request by kjacque**
Add some detail to the high-level security model overview
in the public DAOS docs.

Skip-build: true
Skip-test: true

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

File: doc/overview/security.md [hunk 1]
```diff
@@ -1,14 +1,61 @@
...
+authenticated by means of a certificate generated for that component. These
+components identify one another over the DAOS management network via
+mutually-authenticated TLS.
 
 <a id="4.4.2"></a>
 ## Authorization
 
-DAOS supports a subset of the NFSv4 ACLs for both pools and containers through the properties API.
+Client authorization for resources is controlled by the Access Control List
+(ACL) on the resource, while authorization on the mamagement network is
```
Comment (mjmac) [inline doc/overview/security.md]: typo: mamagement

File: doc/overview/security.md [hunk 2]
```diff
@@ -1,14 +1,61 @@
...
+achieved by settings on the certificates generated while setting up the DAOS
```
Comment (mjmac) [inline doc/overview/security.md]: Worth making this a hyperlink to the section on generating certs?

File: doc/overview/security.md [hunk 3]
```diff
@@ -1,14 +1,61 @@
...
+
+The permissions of a handle last for the duration of its existence, similar to
+an open file descriptor in a POSIX system. A handle cannot currently be revoked.
+
+### Component Certificates
+
+Authorization of RPCs on the DAOS management network is based on the
+certificates configured for each management component. A subset of management
+RPCs is available to each of the trusted component types. The component type
+is indicated in the CommonName (CN) field of the certificate.
```
Comment (mjmac) [inline doc/overview/security.md]: Suggested edit: Access to DAOS management RPCs is controlled via the CommonName (CN) set in each management component certificate. A given management RPC may only be invoked by a component which connects with the correct certificate.

File: doc/overview/security.md (hunk 1 above)
Comment (kjacque) [reply doc/overview/security.md]: Oops, fixed

File: doc/overview/security.md (hunk 2 above)
Comment (kjacque) [reply doc/overview/security.md]: Good idea, I added a link to the deployment doc section about certificate configuration.

File: doc/overview/security.md (hunk 3 above)
Comment (kjacque) [reply doc/overview/security.md]: That does read more clearly, I'll use your version.

**Pull request merged**